tools/CouchDB_Import_Demographics.php

```diff
@@ -277,10 +277,39 @@ function _generateQuery()
                             $cField.DateWithdrawn";
             }
         }
+
+        // Latest Diagnosis by project
+        $projects = \Utility::getProjectList();
+        foreach ($projects as $projectID => $project) {
+            $projectAlias = \Project::getProjectFromID($projectID)->getAlias();
+            $latestProjDx = "latestDiagnosis_$projectAlias";
+
+            $fieldsInQuery .= ", 
+                $latestProjDx.Diagnosis AS $latestProjDx";
+            $tablesToJoin  .= "
+                LEFT JOIN (
+                    SELECT cde.CandID, Diagnosis 
+                    FROM candidate_diagnosis_evolution_rel cde
+                    JOIN diagnosis_evolution de USING (DxEvolutionID)
+                    JOIN (
+                        SELECT cde2.CandID, MAX(OrderNumber) AS OrderNumber 
+                        FROM diagnosis_evolution de2
+                        JOIN candidate_diagnosis_evolution_rel cde2 
+                        USING (DxEvolutionID)
+                        WHERE de2.ProjectID=$projectID
+                        GROUP BY CandID
+                        ) AS maxOrderNumber ON (maxOrderNumber.CandID=cde.CandID 
+                            AND maxOrderNumber.OrderNumber=de.OrderNumber)
+                    WHERE ProjectID=$projectID
+                ) AS $latestProjDx ON ($latestProjDx.CandID=c.CandID)";
+            $groupBy       .= ", $latestProjDx.Diagnosis";
+        }
+
         $whereClause = " WHERE s.Active='Y' AND c.Active='Y' "
                        ."AND c.Entity_type != 'Scanner'";
 
         $concatQuery = $fieldsInQuery . $tablesToJoin . $whereClause . $groupBy;
+        print_r($concatQuery);
```

Comment:
I think this is a debugging print statement

Reply:
@CamilleBeau resolved!